iota/commands/core/get_balances.py

```diff
@@ -44,19 +45,48 @@ def __init__(self):
                     f.Min(0) |
                     f.Max(100) |
                     f.Optional(default=100),
+
+                'tips':
+                    f.Array | f.FilterRepeater(
+                        f.Required |
+                        Trytes(TransactionHash) |
+                        f.Unicode(encoding='ascii', normalize=False),
+                    )
             },
 
             allow_missing_keys={
-                'threshold',
+                'threshold', 'tips',
             },
         )
 
+    def _apply(self, value):
+        value = super(GetBalancesRequestFilter, self)._apply(
+            value
+        )  # type: dict
+
+        if self._has_errors:
+            return value
+
+        # Remove null search terms.
+        # Note: We will assume that empty lists are intentional.
+        search_terms = {
+            term: query
+            for term, query in iteritems(value)
+            if query is not None
+        }
+
+        return search_terms
+
 
 class GetBalancesResponseFilter(ResponseFilter):
     def __init__(self):
         super(GetBalancesResponseFilter, self).__init__({
             'balances': f.Array | f.FilterRepeater(f.Int),
 
-            'milestone':
-                f.ByteString(encoding='ascii') | Trytes(Address),
+            'references':
+                f.Array | f.FilterRepeater(
+                    f.Required |
+                    Trytes(TransactionHash) |
+                    f.Unicode(encoding='ascii', normalize=False),
+                ),
```

Comment:
❔ Could we remove `f.Required` from the response filter?  Match behaviour of other libs, so that node responses don't generate exceptions.
❔ The `f.Unicode` filter will convert `TransactionHash` values into strings.  This is desirable for request filters (so that the string values can be converted to JSON to send to the node), but probably counter-productive for response filters (the client would likely prefer to work with `TransactionHash` objects rather than strings).  Could we remove it from the response filter?

Apologies for not picking up on this during the first review.